e2e/next/src/next-legacy.test.ts

```diff
@@ -279,4 +279,44 @@ describe('@nx/next (legacy)', () => {
     await killPort(prodServePort);
     await killPort(selfContainedPort);
   }, 600_000);
+
+  it('should support --custom-server flag (swc)', async () => {
+    const appName = uniq('app');
+
+    runCLI(
+      `generate @nx/next:app ${appName} --no-interactive --custom-server --linter=eslint --unitTestRunner=jest`,
+      { env: { NX_ADD_PLUGINS: 'false' } }
+    );
+
+    // Check for custom server files added to source
+    checkFilesExist(`${appName}/server/main.ts`);
+    checkFilesExist(`${appName}/.swcrc.server`);
```

Comment:
Let's use `.server.swcrc` to keep consistent with jest tests using `.spec.swcrc`. For RC files like `.eslintrc` and `.babelrc`, the optional extension usually goes to the end, so `.server` would imply it is the extension.